app/src/main/java/com/monkeyteam/chimpagne/viewmodels/EventViewModel.kt

```diff
@@ -86,6 +85,25 @@ class EventViewModel(
     }
   }
 
+  fun fetchAccounts(onSuccess: () -> Unit = {}, onFailure: (Exception) -> Unit = {}) {
+    if (eventID != null) {
+      _uiState.value = _uiState.value.copy(loading = true)
+      viewModelScope.launch {
+        accountManager.getAccounts(
+            setOf(_uiState.value.ownerId) + _uiState.value.guests.keys + _uiState.value.staffs.keys,
+            {
+              _uiState.value = _uiState.value.copy(accounts = it)
+              onSuccess()
+              _uiState.value = _uiState.value.copy(loading = false)

```

Comment:
you're calling onSuccess before loading = false

Reply:
Small typo, thanks for pointing it out.